### DISCUSSION TITLE
Why TypeScript definition for cache object is not complete?

### DISCUSSION THREAD

**Discussion by ansavchenco** [vercel/swr#2213] [Q&A]
Current TypeScript definition for cache object is the following:
```typescript
export interface Cache<Data = any> {
    get(key: Key): Data | null | undefined;
    set(key: Key, value: Data): void;
    delete(key: Key): void;
}
```
The default cache object (when no provided specified) is however a regular js map with all the map methods on it. I sometimes need to use regular map methods like `keys()` and getting TypeScript complaining that there is no such method (even though there is) is very annoying. 

So, is there a particular reason why `Cache` is typed this way and not as a regular map?

**Comment by koba04** [comment]
SWR uses `Map` as the default cache, but also provides a way to define a custom Cache, so the interface has only methods SWR needed.
In v2, SWR adds `keys()` to the interface, so you can use `keys()` without a TS error.
